src/application/service/Wishlist.service.spec.ts

```diff
@@ -0,0 +1,186 @@
+  let productPortMock: jest.Mocked<ProductDbQueryPort>;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        WishlistService,
+        {
+          provide: 'WishlistGateway',
+          useValue: {
+            addWishlist: jest.fn(),
+            removeWishlist: jest.fn(),
+            getWishlistById: jest.fn(),
+            getUserWishlistByUserId: jest.fn(),
+            getByUserIdAndReceiverName: jest.fn(),
+          },
+        },
+        {
+          provide: 'ProductGateway',
+          useValue: {
+            getWishlistProductsByIds: jest.fn(),
+            getWishlistProductsByReceiverName: jest.fn(),
+          },
+        },
+      ],
+    }).compile();
+
+    service = module.get<WishlistService>(WishlistService);
+    queryPortMock = module.get<jest.Mocked<WishlistDbQueryPort>>('WishlistGateway');
+    commandPortMock = module.get<jest.Mocked<WishlistDbCommandPort>>('WishlistGateway');
+    productPortMock = module.get<jest.Mocked<ProductDbQueryPort>>('ProductGateway');
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  describe('addWishlist', () => {
+    it('위시리스트에 상품을 추가할 수 있다', async () => {
+      //given
+      const command = {
+        userId: 1,
+        productId: 101,
+        receiverName: '민수',
+      };
+
+      //when
+      await service.addWishlist(command);
+
+      //then
+      expect(commandPortMock.addWishlist).toHaveBeenCalledWith(command);
+    });
+  });
+
+  describe('removeWishlist', () => {
+    it('자신의 위시리스트에서 상품을 삭제할 수 있다', async () => {
+      //given
+      queryPortMock.getWishlistById.mockResolvedValue(domainWishlistMockData);
+
+      //when
+      await service.removeWishlist(domainWishlistMockData.id, domainWishlistMockData.userId);
+
+      //then
+      expect(commandPortMock.removeWishlist).toHaveBeenCalledWith(domainWishlistMockData.id);
+      expect(commandPortMock.removeWishlist).toHaveBeenCalledTimes(1);
+    });
+    it('존재하지 않는 위시리스트 ID일 경우 예외를 던진다', async () => {
+      // given
+      const wishlistId = 999;
+      const userId = 1;
+      commandPortMock.removeWishlist.mockImplementation(() => {
+        throw new WishlistNotFoundException();
+      });
+
+      //  when & then
+      await expect(service.removeWishlist(wishlistId, userId)).rejects.toThrow(
+        WishlistNotFoundException,
+      );
+    });
+    it('다른 사용자의 위시리스트일 경우 예외를 던진다', async () => {
+      // given
+      const wishlistOwnerId = 999;
+      queryPortMock.getWishlistById.mockResolvedValue({
+        ...domainWishlistMockData,
+        userId: wishlistOwnerId,
+      });
+
+      // when & then
+      const requestUserId = 1;
+      await expect(
+        service.removeWishlist(domainWishlistMockData.id, requestUserId),
+      ).rejects.toThrow(ForbiddenWishlistAccessException);
+    });
+  });
+  describe('getWishlistGroupsByUserId', () => {
```

Comment:
메소드 순서는 서비스에 정의된 것과 동일하게 get->create->update->delete순으로 배열 부탁드립니다
```suggestion
  });
  
  describe('getWishlistGroupsByUserId', () => {
```

Reply:
수정하였습니다!